File: examples/lh-server/python/06-threads/tasks.py
from littlehorse.worker import WorkerContext


async def my_task(input: str, context: WorkerContext) -> str:
    thread_run_number = context.node_run_id.thread_run_number
    thread_name = "parent" if thread_run_number == 0 else "child"
    result = f"Hello from the {thread_name} thread: {input}"
    print(result)
    return result

File: examples/lh-server/go/06-threads/tasks.go
package main

import "github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"

func MyTask(input string, context *littlehorse.WorkerContext) string {
	threadName := "child"
	if context.GetNodeRunId().GetThreadRunNumber() == 0 {
		threadName = "parent"
	}
	return "Hello from the " + threadName + " thread: " + input
}
